### DISCUSSION TITLE
restart with TensorMechanics (reloading stress state without interpolating)

### DISCUSSION THREAD

**Discussion by Thomas-Ulrich** [idaholab/moose#25460] [Q&A General] [answered]
Hi,

We are running a simulation based on a (THM) moose application, which includes the tensor mechanics module.
E.g., we have the following block in our input file:
```
[Modules]
  [./TensorMechanics]
    [./Master]
      [./all]
      add_variables = true
      strain = SMALL
      incremental = true
      eigenstrain_names = 'reduced_eigenstrain'
      volumetric_locking_correction = true
      additional_generate_output = 'stress_xx stress_xy stress_xz stress_yy stress_yz stress_zz'
      [../]
    [../]
  [../]
[]
```
When we reach some critical state in the simulation, we want to extract the stresses from Moose, use them in another code, update the stresses and restart a new moose simulation. Let's forget about the update of the stress, but focus on the loading of the previous stress state in a new moose simulation. We are currently doing it using a SolutionUserObject reading the exodus output file from the first simulation.

```
[UserObjects]
  [./solution_loop]   
    type = SolutionUserObject
    mesh = output.e      # should be the name of output in first loop
    timestep = LATEST
  [../]
[]
[Functions]
  [sxx_functionT]
    type = SolutionFunction
    from_variable = 'stress_xx'
    solution = 'solution_loop'
    execute_on = 'INITIAL'    
 []
```
And we use sxx_functionT to initialize the new simulation. But this does not work perfectly. The new stress differs slightly (interpolation) from the last time step of the first simulation. 

Our mesh is 3D unstructured with lower dimensional elements (2D fault, and 1D wells).
We think it does not work perfectly because probably the SolutionUserObject interpolate stress_xx from the volume element 
onto the 2D fault, instead of loading the 2D data from the exodus file.
Therefore, it would be better to load the stress from the exodus file using checkpointing rather than  SolutionUserObject (e.g. using `initial_from_file_timestep = xx`). But how can we do that, given that the variables stress_ij are not defined anywhere in our input file, except in the TensorMechanics block copied above?

Thanks in advance for your help,

Thomas Ulrich.

**Comment by GiudGiud** [comment, marked as answer]
Hello

To use checkpoint you ll want the system to be exactly the same size before and after the restart. 
If that is so, you can set "restart_file_base" in the Problem block, and it will restart everything it can.

You can also use a SolutionUO with a checkpoint file, but I think it will always interpolate

Guillaume

**Reply by Thomas-Ulrich** [reply]
thank you I will try
